crates/oxc_linter/src/rules/import/no_mutable_exports.rs

```diff
@@ -0,0 +1,244 @@
+use oxc_ast::{
+    AstKind,
+    ast::{
+        Declaration, Expression, ModuleExportName, VariableDeclaration, VariableDeclarationKind,
+    },
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::ReferenceId;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule};
+
+fn no_mutable_exports_diagnostic(span: Span, kind: VariableDeclarationKind) -> OxcDiagnostic {
+    let kind_str = if kind == VariableDeclarationKind::Var { "var" } else { "let" };
+    OxcDiagnostic::warn(format!("Exporting mutable '{kind_str}' binding, use 'const' instead."))
+        .with_label(span)
+}
+
+/// <https://github.com/import-js/eslint-plugin-import/blob/v2.31.0/docs/rules/no-mutable-exports.md>
+#[derive(Debug, Default, Clone)]
+pub struct NoMutableExports;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Forbids the use of mutable exports with var or let.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// In general, we should always export constants
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// export let count = 2
+    /// export var count = 3
+    ///
+    /// let count = 4
+    /// export { count }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// export const count = 1
+    /// export function getCount() {}
+    /// export class Counter {}
+    /// ```
+    ///
+    /// ### Functions/Classes
+    /// Note that exported function/class declaration identifiers may be reassigned,
+    /// but are not flagged by this rule at this time. They may be in the future.
+    NoMutableExports,
+    import,
+    style,
+);
+
+impl Rule for NoMutableExports {
+    fn run_once(&self, ctx: &LintContext) {
+        for node in ctx.semantic().nodes() {
+            match node.kind() {
+                AstKind::ExportNamedDeclaration(export_name_decl) => {
```

Comment:
using `ctx.semantic().nodes()` is slow as it loops over every single node. can we just move this into `fn run`, and use the existing loop?